### ISSUE TITLE
error returns of engine pointers

### ISSUE THREAD

**Issue by DanGrayson** [Macaulay2/M2#486]
Many functions in engine.h are declared to return a pointer when returning a bool would 
be better, e.g.:
```
Computation /* or null */*
rawStartComputation(Computation *C)
```
Let's make them all return bool.  The point is that it makes the code in interface.dd counterintuitive, for the interface routine ends up with two pointers to the same thing.

**Comment by mikestillman** [comment]
How should this look then? e.g:
Matrix* /* or null */ rawMyMatrixFunction(long n, const Matrix* A)
should become:
bool rawMyMatrixFunction(long n, const Matrix* A, Matrix*& result) ?

**Comment by DanGrayson** [comment]
No, that type of thing is okay, but for `rawStartComputation` there is no return value needed, except the error indication, as the computation returned is the same as the computation provided.
